tests/core/pyspec/eth2spec/test/phase0/unittests/fork_choice/test_on_block.py

```diff
@@ -1,239 +1,47 @@
-    block = build_empty_block_for_next_slot(spec, state)
-    transition_unsigned_block(spec, state, block)
-    block.state_root = state.hash_tree_root()
-    store = spec.get_forkchoice_store(state, block)
-    store.finalized_checkpoint = spec.Checkpoint(
-        epoch=store.finalized_checkpoint.epoch + 2,
-        root=store.finalized_checkpoint.root
-    )
-
-    # First transition through the epoch to ensure no skipped slots
-    state, store, _ = apply_next_epoch_with_attestations(spec, state, store)
-
-    # Now build a block at later slot than finalized epoch
-    # Includes finalized block in chain, but not at appropriate skip slot
-    block = build_empty_block(spec, state, spec.compute_start_slot_at_epoch(store.finalized_checkpoint.epoch) + 2)
-    signed_block = state_transition_and_sign_block(spec, state, block)
-    spec.on_tick(store, store.time + state.slot * spec.config.SECONDS_PER_SLOT)
-    run_on_block(spec, store, signed_block, False)
-
-
-@with_all_phases
-@spec_state_test
-def test_on_block_update_justified_checkpoint_within_safe_slots(spec, state):
-    # Initialization
-    store = get_genesis_forkchoice_store(spec, state)
-    time = 0
-    spec.on_tick(store, time)
-
-    next_epoch(spec, state)
-    spec.on_tick(store, store.time + state.slot * spec.config.SECONDS_PER_SLOT)
-    state, store, last_signed_block = apply_next_epoch_with_attestations(spec, state, store)
-    next_epoch(spec, state)
-    spec.on_tick(store, store.time + state.slot * spec.config.SECONDS_PER_SLOT)
-    last_block_root = hash_tree_root(last_signed_block.message)
-
-    # Mock the justified checkpoint
-    just_state = store.block_states[last_block_root]
-    new_justified = spec.Checkpoint(
-        epoch=just_state.current_justified_checkpoint.epoch + 1,
-        root=b'\x77' * 32,
-    )
-    just_state.current_justified_checkpoint = new_justified
-
-    block = build_empty_block_for_next_slot(spec, just_state)
-    signed_block = state_transition_and_sign_block(spec, deepcopy(just_state), block)
-    assert spec.get_current_slot(store) % spec.SLOTS_PER_EPOCH < spec.SAFE_SLOTS_TO_UPDATE_JUSTIFIED
-    run_on_block(spec, store, signed_block)
-
-    assert store.justified_checkpoint == new_justified
+from eth2spec.utils.ssz.ssz_impl import hash_tree_root
+from eth2spec.test.context import (
+    spec_state_test,
+    with_all_phases,
+)
+from eth2spec.test.helpers.block import (
+    build_empty_block_for_next_slot,
+)
+from eth2spec.test.helpers.fork_choice import (
+    get_genesis_forkchoice_store,
+    run_on_block,
+    apply_next_epoch_with_attestations,
+)
+from eth2spec.test.helpers.state import (
+    next_epoch,
+    state_transition_and_sign_block,
+)
 
 
 @with_all_phases
 @spec_state_test
 def test_on_block_outside_safe_slots_and_multiple_better_justified(spec, state):
+    """
+    NOTE: test_new_justified_is_later_than_store_justified also tests best_justified_checkpoint
+    """
```

Comment:
To reviewers: I'm inclined to not translate this test case since we can test `best_justified_checkpoint` in other ways. Please let me know if I miss anything. 🙏